Simplified/Reader2/Bookmarks/NYPLAnnotations.swift

```diff
@@ -494,21 +498,28 @@ protocol NYPLAnnotationSyncing: AnyObject {
                                       motivation: NYPLBookmarkSpec.Motivation,
                                       publication: Publication?,
                                       bookID: String) -> [NYPLReadiumBookmark]? {
+    let metadata: [String: Any] = ["bookID":bookID]
     if let error = error as NSError? {
-      Log.error(#file, "Request Error Code: \(error.code). Description: \(error.localizedDescription)")
+      NYPLErrorLogger.logError(error,
+                               summary: "NYPLAnnotations::parseAnnotationsResponse error",
+                               metadata: metadata)
       return nil
     }
 
     guard let data = data,
       let jsonObject = try? JSONSerialization.jsonObject(with: data, options: []),
       let json = jsonObject as? [String: Any] else {
-        Log.error(#file, "Response from annotation server could not be serialized.")
+        NYPLErrorLogger.logError(withCode: .serializationFail,
+                                 summary: "NYPLAnnotations::parseAnnotationsResponse error",
+                                 metadata: metadata)
         return nil
     }
 
     guard let first = json["first"] as? [String: Any],
       let items = first["items"] as? [[String: Any]] else {
-        Log.error(#file, "Missing required key from Annotations response, or no items exist.")
+        NYPLErrorLogger.logError(withCode: .annotationFeedNoData,
```

Comment:
curious to know what you think about this. So, we currently have a `opdsFeedNoData` error code. What if instead of creating a new code (`annotationFeedNoData`) for the same type of error we just had a single `noData` code? I think it will be just as informative because the `summary` of these errors will provide the needed context anyway. A single `noData` code will allow us to slice the Crashlytics data orthogonally and easily find "all the instances where we obtained "no data" where we expected to. You can see this in action for example for the `900` error code ("no URL")

<img width="624" alt="900" src="https://user-images.githubusercontent.com/289104/154158959-a52edf5f-8f54-4fba-aa7b-f937699a92ca.png">

I'd be in favor but let me know what you think.

Reply:
Yeah this sounds right to me, updated